integration_tests/src/main/python/fastparquet_compatibility_test.py

```diff
@@ -76,25 +76,38 @@ def read_with_fastparquet_or_plugin(spark):
                               start=datetime(1, 1, 1, tzinfo=timezone.utc),
                               end=datetime(1899, 12, 31, tzinfo=timezone.utc)),
                  marks=pytest.mark.xfail(reason="fastparquet reads timestamps preceding 1900 incorrectly.")),
-    #  TODO: Array gen type deduction borked when converting from Pandas to Spark dataframe.
-    # ArrayGen(child_gen=IntegerGen(nullable=False), nullable=False),
-    #  TODO: Struct rows seem to be correct, but are failing comparison because of differences in Row representation.
-    # StructGen(children=[("first", IntegerGen(nullable=False)),
-    #                     ("second", FloatGen(nullable=False))], nullable=False)
+    pytest.param(
+        ArrayGen(child_gen=IntegerGen(nullable=False), nullable=False),
+        marks=pytest.mark.xfail(reason="Conversion from Pandas dataframe to Spark dataframe fails: "
```

Comment:
Do we know why it is failing? It could not infer the types. I see APIs in Spark that allow us to pass in a schema. Should we do that?

https://github.com/apache/spark/blob/39d43e0ac3b58fb7e804362bb07665e8d6536250/python/pyspark/sql/session.py#L1227-L1232

Reply:
```python
   >>> sql(" values (array(1,2,3)), (array(4,5,6)) as foo(arr) ").write.mode("overwrite").parquet("/tmp/foobar")
   >>> pdf = fastparquet.ParquetFile("/tmp/foobar").to_pandas()
   >>> spark.createDataFrame(pdf)
   ...
   TypeError: Unable to infer the type of the field arr.
   ```
   Specifying a schema does not seem to help matters.
   ```python
   >>> spark.createDataFrame(pdf, schema=StructType([StructField('arr', ArrayType(IntegerType(), True), True)]))
   ...
   TypeError: element in array field arr: IntegerType() can not accept object 1 in type <class 'numpy.int32'>
   ```